packages/mobile/src/screens/edit-track-screen/screens/RemixSettingsScreen.tsx

```diff
@@ -107,11 +99,10 @@ export const RemixSettingsScreen = () => {
   const [isRemixUrlMissing, setIsRemixUrlMissing] = useState(false)
   const [isTouched, setIsTouched] = useState(false)
   const navigation = useNavigation()
-  const dispatch = useDispatch()
-  const parentTrack = useSelector(getTrack)
-  const parentTrackArtist = useSelector(getUser)
-  const parentTrackStatus = useSelector(getStatus)
-  const isInvalidParentTrack = parentTrackStatus === Status.ERROR
+
+  const permalink = useThrottle(getPathFromTrackUrl(remixOfInput), 1000)
+  const { data: parentTrack } = useTrackByPermalink(permalink)
```

Comment:
does react-query have throttling built in? or are we going to make a failed request for each letter of the input?

Reply:
good question, so the permalink itself is throttled, so we will make throttled failed requests.